scala-thoughts.md

```diff
@@ -0,0 +1,9 @@
+I'm not really sure if I have a judgement yet in the sense of whether or not I "like" Scala. Overall, I have a few mixed opinions, and some initial impressions, but I have intentionally tried not to form an opinion too soon, because I think I might be slightly biased by some external factors. Primarily, Scala was a huge pain to install and use (on Windows), so I'm not sure how I'd actually set up a scala project or script on my own. Second, and relatedly, I really dislike eclipse - its slow, terrible to look at, and has really unhelpful error messages sometimes. Since these things aren't actually issues with the language itself, I don't want to form an impression too soon.
+
+With that said, there are some concrete things I noticed that I can say. Firstly, Scala's syntax is pretty weird. The fact that you can do things in multiple ways is a little bit concerning to me. The best example is the fact that you can write functions with either the "dot" notation, or with spaces between them. In some cases, I can see the spaces notation being very elegant, but when you have lots of arguments, with complex chains of higher-order functions, I really like the dot notation for clarifying exactly what function is calling what. I worry that its too easy to adopt one or the other as a standard, which could lead to confusing code, or mixing the two, which leads to inconsistent code. I tried on the assignment to vary which ones i used, but overall the difference seemed to be pretty trivial unless the spaces notation was too confusing to use.
```

Comment:
I said the flexibility in syntax was kind of nice, but I qualified my statement with pretty much the exact same sentiment. It was nice for me in the sense that I could kind of type whatever and it would compile, even when I wasn't exactly sure what I was doing. I specifically mentioned how it could get kind of problematic with a significantly larger coding base as I'm not sure whether there is a quintessential Scala elegance guide.